File: get.extrema_max.pl
#!/usr/bin/perl

$DIFAX_MAX_PATH = "/home/disk/data/archive/images/difax/difax_max";

$ARCHIVE_PATH = "/home/disk/funnel/impacts/archive/ops/extrema";

# Look for yesterday's map

$month = `date --date='yesterday' "+%Y%m"`;
chop $month;
#print "month = $month\n";
$yesterday = `date --date='yesterday' "+%Y%m%d"`;
chop $yesterday;
#print "yesterday = $yesterday\n";
$yesterdayDir = $ARCHIVE_PATH."/".$yesterday;
unless (-e $yesterdayDir) {
    mkdir $yesterdayDir;
}

chdir($DIFAX_MAX_PATH."/".$month);
#print "changed to $DIFAX_MAX_PATH/$month\n";
foreach $file (<$yesterday*0000.gif>) {
  print "file = $file\n";
  ($dateTime) = ($file =~ /(.*)\.gif/);
  $date = substr($dateTime,0,8);
  #print "date = $date\n";
  $outFile = $ARCHIVE_PATH."/".$date."/ops.extrema.".$dateTime.".max_temp.gif";
  print "outFile = $outFile\n";
  unless (-e $outFile) {
    system("/bin/cp $file $outFile");
  }
}

# Look for today's map

$month = `date "+%Y%m"`;
chop $month;
#print "month = $month\n";
$today = `date "+%Y%m%d"`;
chop $today;
#print "today = $today\n";
$todayDir = $ARCHIVE_PATH."/".$today;
unless (-e $todayDir) {
    mkdir $todayDir;
}

chdir($DIFAX_MAX_PATH."/".$month);
#print "changed to $DIFAX_MAX_PATH/$month\n";
foreach $file (<$today*0000.gif>) {
  print "file = $file\n";
  ($dateTime) = ($file =~ /(.*)\.gif/);
  $date = substr($dateTime,0,8);
  #print "date = $date\n";
  $outFile = $ARCHIVE_PATH."/".$date."/ops.extrema.".$dateTime.".max_temp.gif";
  print "outFile = $outFile\n";
  unless (-e $outFile) {
    system("/bin/cp $file $outFile");
  }
}

exit(0);

File: get_extrema_max.py
#!/usr/bin/python3

import os
import time
from datetime import date
from datetime import datetime
from datetime import timedelta
import shutil
from ftplib import FTP

test = False
#difaxMaxDir = '/home/disk/data/archive/images/difax/difax_max'
difaxMaxDir = '/home/disk/data/images/difax/difax_max'
# Possible replacement:
# https://www.cpc.ncep.noaa.gov/products/tanal/1day/max_min/20211214.1day.max_min.C.gif
tempDir = '/tmp'
catalogPrefix = 'surface.NWS'
    
# Field Catalog inputs
if test:
    ftpCatalogServer = 'ftp.atmos.washington.edu'
    ftpCatalogUser = 'anonymous'
    ftpCatalogPassword = 'brodzik@uw.edu'
    catalogDestDir = 'brodzik/incoming/impacts'
else:
    ftpCatalogServer = 'catalog.eol.ucar.edu'
    ftpCatalogUser = 'anonymous'
    catalogDestDir = '/pub/incoming/catalog/impacts'

# Open ftp connection to NCAR sever
if test:
    catalogFTP = FTP(ftpCatalogServer,ftpCatalogUser,ftpCatalogPassword)
    catalogFTP.cwd(catalogDestDir)
else:
    catalogFTP = FTP(ftpCatalogServer,ftpCatalogUser)
    catalogFTP.cwd(catalogDestDir)

# Look for yesterday's map
today = datetime.utcnow()
yesterday = today - timedelta(days = 1)
dateStr = yesterday.strftime("%Y%m%d")

# note for 2021: new maps are in DifaxMaxDir+'/'+year+month
# e.g. /home/disk/data/archive/images/difax/difax_max/202112
#for file in os.listdir(os.path.join(difaxMaxDir,yearMonthStr)):
for file in os.listdir(difaxMaxDir):
    if dateStr in file and file.endswith('gif'):
        print('file = ',file)
        (dateTime,ext) = file.split('.')
        date = dateTime[0:8]
        print('date = ',date)
        catalogName = catalogPrefix+'.'+dateTime+'.max_temp.gif'
        print('catalogName = ',catalogName)
    
        #shutil.copy(os.path.join(difaxMaxDir,yearMonthStr,file),
	#	    tempDir+'/'+catalogName)
        shutil.copy(os.path.join(difaxMaxDir,file),
		    tempDir+'/'+catalogName)
            
        # ftp file to catalog location
        ftpFile = open(os.path.join(tempDir,catalogName),'rb')
        catalogFTP.storbinary('STOR '+catalogName,ftpFile)
        ftpFile.close()
        
        # remove file from tempDir
        os.remove(os.path.join(tempDir,catalogName))

# Look for today's map
dateStr = today.strftime("%Y%m%d")

#for file in os.listdir(os.path.join(difaxMaxDir,yearMonthStr)):
for file in os.listdir(difaxMaxDir):
    if dateStr in file and file.endswith('gif'):
        print('file = ',file)
        (dateTime,ext) = file.split('.')
        date = dateTime[0:8]
        print('date = ',date)
        catalogName = catalogPrefix+'.'+dateTime+'.max_temp.gif'
        print('catalogName = ',catalogName)
    
        #shutil.copy(os.path.join(difaxMaxDir,month,file),
	#	    tempDir+'/'+catalogName)
        shutil.copy(os.path.join(difaxMaxDir,file),
		    tempDir+'/'+catalogName)
            
        # ftp file to catalog location
        ftpFile = open(os.path.join(tempDir,catalogName),'rb')
        catalogFTP.storbinary('STOR '+catalogName,ftpFile)
        ftpFile.close()

        # remove file from tempDir
        os.remove(os.path.join(tempDir,catalogName))

# Close ftp connection
catalogFTP.quit()
